corva/network/api.py

```diff
@@ -1,60 +1,75 @@
 
 
-@dataclass(eq=False)
 class Api:
-    HTTP_METHODS = {'GET', 'POST', 'PATCH', 'PUT', 'DELETE'}
-
-    timeout: int = 600  # seconds
-    max_retries: int = 3
-    api_url: str = settings.API_ROOT_URL
-    data_api_url: str = settings.DATA_API_ROOT_URL
-    api_key: str = settings.API_KEY
-    app_name: str = settings.APP_NAME
-    session: Session = field(default_factory=Session)
-
-    def __post_init__(self):
-        self.session.headers.update({
-            'Authorization': f'API {self.api_key}',
-            'X-Corva-App': self.app_name
+    ALLOWED_METHODS = {'GET', 'POST', 'PATCH', 'PUT', 'DELETE'}
+
+    def __init__(
+         self,
+         api_url: str = settings.API_ROOT_URL,
+         data_api_url: str = settings.DATA_API_ROOT_URL,
+         api_key: str = settings.API_KEY,
+         app_name: str = settings.APP_NAME,
+         timeout: int = 600,
+         max_retries: int = 3
+    ):
+        self.timeout = timeout
+        self.max_retries = max_retries
+        self.api_url = api_url
+        self.data_api_url = data_api_url
+        self.api_key = api_key
+        self.app_name = app_name
+        self.session = self._init_session(
+            api_key=api_key, app_name=app_name, max_retries=max_retries, allowed_methods=list(self.ALLOWED_METHODS)
+        )
+
+    @staticmethod
+    def _init_session(api_key: str, app_name: str, max_retries: int, allowed_methods: List[str]):
+        session = Session()
+
+        session.headers.update({
+            'Authorization': f'API {api_key}',
+            'X-Corva-App': app_name
         })
-        self.session.mount(
+        session.mount(
             'https://',
             HTTPAdapter(
                 max_retries=Retry(
-                    total=self.max_retries,
+                    total=max_retries,
                     status_forcelist=[408, 429, 500, 502, 503, 504],
-                    allowed_methods=list(self.HTTP_METHODS),
+                    allowed_methods=allowed_methods,
                     backoff_factor=0.3,
                     raise_on_status=False
                 )
             )
         )
 
-    def get(self, path, **kwargs):
+        return session
+
+    def get(self, path: str, **kwargs):
         return self._request('GET', path, **kwargs)
 
-    def post(self, path, **kwargs):
+    def post(self, path: str, **kwargs):
         return self._request('POST', path, **kwargs)
 
-    def patch(self, path, **kwargs):
+    def patch(self, path: str, **kwargs):
         return self._request('PATCH', path, **kwargs)
 
-    def put(self, path, **kwargs):
+    def put(self, path: str, **kwargs):
         return self._request('PUT', path, **kwargs)
 
-    def delete(self, path, **kwargs):
+    def delete(self, path: str, **kwargs):
         return self._request('DELETE', path, **kwargs)
 
-    def _get_url(self, path):
+    def _get_url(self, path: str):
         if path.startswith(self.api_url) or path.startswith(self.data_api_url):
```

Comment:
user already passed `api_url` and `data_api_url` into the constructor of this class. Why do we need to check if he is using the correct path? By possessing these values we have control over them and it's better to use them and not burden a user to provide a path in the format `f"{API_URL}/entity"`.

Though, I see a bigger problem here. It's unclear how the user will be using this class

Reply:
Yeah, will delete this logic, it is obsolete.
User is supposed to use the class like this
```python
api = Api()
response = api.get(path='/api/v1/data/corva/dataset/')
```

Reply:
We would like to take responsibility over switching api version too. So, if we are going update version - we need to roll-out new lib version